### DISCUSSION TITLE
Plotting flag variables

### DISCUSSION THREAD

**Discussion by maxwelllevin** [ARM-DOE/ACT#471] [Ideas]
Hi act-atmos team, 

I'm working with some ARM data that has a state indicator variable I want to plot. In my case I am working with there is a `cloud_phase` variable which looks something like this:

```
cloud_phase(time, height):byte*
    long_name = Thermodynamic Cloud Phase
    units = 1
    flag_values:byte = 0, 1, 2, 3, 4, 5, 6, 7, 8
    flag_meanings = clear_sky liquid ice mixed_phase drizzle liquid_drizzle rain snow unknown
    missing_value:byte = -1
    _FillValue:byte = -1
```

So a value of 0 indicates clear sky, 1 indicates liquid, 2 indicates ice, etc. Note this is a bit different than QC variables because the value of `cloud_phase` will always match one of the `flag_values` or `_FillValue` exactly, whereas qc variables can match multiple `flag_masks` / bits to indicate a combination of different states. In other words the different states of the `cloud_phase` variable are mutually exclusive.

I want the plot to look something like this:
<img width="1199" alt="image" src="https://user-images.githubusercontent.com/24307537/171961868-688bcb04-7c3b-4efb-9ac3-60fda851152c.png">

To the best of my knowledge, `act` doesn't quite do this yet (though it does do superbly with plotting qc variables). It would certainly be beneficial to me if the `act.plotting` module could be expanded to provide a method to plot mutually exclusive state indicator variables, so I wanted to float the idea to you all and see if you think it's worthwhile.

## Reference

ARM Standards §6.5 discusses how ARM handles exclusive and inclusive state indicator variables:
https://www.arm.gov/publications/programdocs/doe-sc-arm-15-004.pdf

The CF Conventions also discuss variable flags here:
https://cfconventions.org/cf-conventions/cf-conventions.html#flags

**Comment by AdamTheisen** [comment]
@maxwellevin this is a great idea and I'm wondering if we can fit it in with the existing plotting with some discrete shading keywords.  I'm going to create an issue from this and add it to the list.

**Comment by kenkehoe** [comment]
@maxwellevin what datastream did you use for your example plot?

**Reply by maxwelllevin** [reply]
It's `thermocldphase`, a new ARM vap that hasn't been released for evaluation yet (coming soon). I can point you to some data on the ARM server if you'd like to use part of it to test.

**Comment by kenkehoe** [comment]
I've created PR #480 for this feature request


![test_y_axis_flag_meanings](https://user-images.githubusercontent.com/8261054/173927344-50e614ca-4611-4b6d-a44a-5724502469d6.png)
![test_colorbar_lables](https://user-images.githubusercontent.com/8261054/173927647-b70343af-b083-4b8e-b94a-6e947a53e9ef.png)
.
